docs/provisioning-state-must-be-read-only.md

```diff
@@ -8,16 +8,168 @@ ARM Error
 
 ARM OpenAPI(swagger) specs
 
-## Output Message
+## Related ARM Guideline Code
 
-provisioningState property must be set to readOnly.
+- RPC-Async-V1-16
 
 ## Description
 
 This is a rule introduced to validate if provisioningState property is set to readOnly.
 
 ## How to fix the violation
 
-Set the `provisioningState` property `readOnly`.
-I.e., `"readOnly": true`
-should be added to "provisioningState" property.
+Set the `provisioningState` property `readOnly`, i.e, `"readOnly": true` should be added to "provisioningState" property.
+Make sure to add the `"readOnly": true` to the actual "provisioningState" definition and not to the envelope property(i.e, property that references the actual definition)
+
+
+## Bad example 1 
+
+provisioningState without readOnly 
+
+```json5
+ "put": {
+    "responses": {
+      "200": {
+        "description": "Success",
+        "schema": {
+          "$ref": "#/definitions/FooProps",
+        },
+      },
+    }
+ }
+"definitions": {
+    "FooProps": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "Provisioning state of the foo rule.",
+          "enum": ["Creating", "Canceled", "Deleting", "Failed"],
+        },
+      },
+    },
+}
+```
+
+    
+## Bad example 2 
+
+provisioningState with readOnly defined in the envelope property and not in the actual definition
+
+```json5
+ "put": {
+    "responses": {
+      "200": {
+        "description": "Success",
+        "schema": {
+          "$ref": "#/definitions/FooRule",
+        },
+      },
+    }
+ }
+"definitions": {
+   "ProvisioningState": {
+      "type": "string",
+      "enum": ["Succeeded", "Failed", "Canceled"],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true,
+      },
+    },
+    "FooRule": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FooProps",
+          "x-ms-client-flatten": true,
+        },
+      },
+    },
+    "FooProps": {
+      "properties": {
+        "provisioningState": {
+        "$ref": "#/definitions/ProvisioningState",
+        "readOnly": true,
+        "type": "string",
+        "description": "Provisioning state of the foo rule.",
+        "enum": ["Creating", "Canceled", "Deleting", "Failed"],
```

Comment:
pl remove these 3 lines for type, description and enum. This example should only have $ref and readOnly proeprties

Reply:
I thought, it could have the actual def. here